### PR TITLE
Fix glob patterns with mixing relative and absolute paths.

### PR DESCRIPTION
**Pull request by Dominator008**
See https://groups.google.com/forum/#!topic/closure-compiler-discuss/vIkk0XImMpg.

Also, skip a file or directory if visiting it failed.

Wrap all renameTo() calls with assertTrue() in CommandLineRunnerTest.

[Review on Reviewable](https://reviewable.io/reviews/google/closure-compiler/1619)

### CODE REVIEW

**Comment by Dominator008** [comment]
It's kind of hard to write a unit test for this that works internally :(

**Comment by MatrixFrog** [comment]
If you write one that only works externally then we can do some MOE trickery so that it doesn't run internally. Or if it makes more sense, maybe the test can just start with essentially `if (!WINDOWS) {return}`.

**Comment by Dominator008** [comment]
@MatrixFrog It's not Windows specific. I added two tests with `// NOTE(moz): This might not work on Google's internal build system.`.

**Comment by MatrixFrog** [comment]
Okay, well I'll give it a try.

**Comment by Dominator008** [comment]
@MatrixFrog IIRC @blickly said creating files in the current directory worked internally, but not creating directories. I modified the tests to not create directories so they might actually work internally.

File: test/com/google/javascript/jscomp/CommandLineRunnerTest.java [hunk 1]
```diff
@@ -1213,6 +1214,57 @@ public void testGlobJs5() throws IOException, FlagUsageException {
...
+    // Make sure patterns like "!**\./ignored**.js" work
+    String glob1 = "!**\\." + File.separator + "ignored**.js";
+    String glob2 = new File(jsFile1.value).getParent() + File.separator + "**.js";
+    compileFiles(
+        "var a;", new FlagEntry<>(JsSourceType.JS, glob1),
+        new FlagEntry<>(JsSourceType.JS, glob2));
+    ignoredJs.delete();
+  }
+
+  // NOTE(moz): This might not work on Google's internal build system.
```
Comment (MatrixFrog) [inline test/com/google/javascript/jscomp/CommandLineRunnerTest.java]: It seems to work. Will review internally.

Comment (Dominator008) [inline test/com/google/javascript/jscomp/CommandLineRunnerTest.java]: Nice, thanks :)

**Comment by MatrixFrog** [comment]
Oops, I was wrong. testGlobJs7 is failing but I don't see why. Anyway we'll figure it out.

**Comment by Dominator008** [comment]
@MatrixFrog Might be a permission issue, but in that case testGlobJs6() should also fail.

**Comment by MatrixFrog** [comment]
It is a permissions issue, and if you wrap all your `renameTo` calls in an `assertTrue` then the other test fails too. https://docs.oracle.com/javase/7/docs/api/java/io/File.html#renameTo(java.io.File) recommends this.

**Comment by Dominator008** [comment]
@MatrixFrog Alright.. I wrapped all of the `renameTo` calls with `assertTrue`. We'd have to do some MOE trickery with `testGlobJs6` and `testGlobJs7` then :(

**Comment by MatrixFrog** [comment]
Ran into another problem: This breaks the use of whitelist files. The whitelist file will generally use a path that is relative to the root of the project, but the error is reported using the absolute path, so the whitelist fails to suppress the error.

**Comment by Dominator008** [comment]
@MatrixFrog OK I changed to check the normalized absolute paths for exclusion, but return the original paths. Hopefully it should work now.

**Pull request closed without merging**